botocore/utils.py

```diff
@@ -53,6 +52,83 @@
     ConnectTimeoutError,
 )
 S3_ACCELERATE_WHITELIST = ['dualstack']
+# In switching events from using service name / endpoint prefix to service
+# id, we have to preserve compatibility. This maps the instances where either
+# is different than the transformed service id.
+EVENT_ALIASES = {
```

Comment:
What's your thoughts on adding some functional tests for these aliases? I was thinking it would be similar to what we did for `sagemaker` but be more programmatic. Specifically, we would have two tests where one emits events for service names that do not match the service id and the other emits events for endpoint prefixes that do not match the service id. Then in each yielded case we would register a handler to an event that uses the appropriate alias and make sure it gets fired. The list for the cases to run would also be statically generated. I think this can give us a lot of confidence in these aliasings.